### PR TITLE
neomutt: adding unmailboxes option

### PR DESCRIPTION
**Pull request by randomn4me**
### Description

Adding unmailboxes option to neomutt, which adds it to every account.email with neomutt enabled.

Docs: https://www.mutt.org/doc/manual/#mailboxes

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added.

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

### CODE REVIEW

File: modules/programs/neomutt.nix [hunk 1]
```diff
@@ -397,6 +402,12 @@ in {
           default = true;
         };
 
+      unmailboxes = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Set 'unmailboxes *' at the start of account configs";
```
Comment (rycee) [inline modules/programs/neomutt.nix:408]:
```suggestion
        description = "Set `unmailboxes *` at the start of account configs.";
```

File: modules/programs/neomutt.nix [hunk 2]
```diff
@@ -397,6 +402,12 @@ in {
           default = true;
         };
 
+      unmailboxes = mkOption {
```
Comment (rycee) [inline modules/programs/neomutt.nix:405]: Would this name be easily understood by a neomutt user? I wonder if perhaps a more descriptive option name could be used?

Or at least expand the option description with a sentence on what the actual effect the option has.

**Comment by rycee** [comment]
Oh, and about the test error. It should be resolved if you rebase your branch on the latest master.

File: modules/programs/neomutt.nix (hunk 2 above)
Comment (randomn4me) [reply modules/programs/neomutt.nix:405]: I will add more information to this option as I learned about this option only after scouring the mutt manual :+1:

**Comment by randomn4me** [comment]
Updaten everything and it should now work :+1:

**Comment by rycee** [comment]
Thanks! Merged to master now 🙂

**Pull request merged**